src/schema/schema_create.c

```diff
@@ -961,6 +975,102 @@ __create_data_source(
     return (dsrc->create(dsrc, &session->iface, uri, (WT_CONFIG_ARG *)cfg));
 }
 
+/*
+ * __import_entry_cmp --
+ *     Qsort function: sort the import entries array by name.
+ */
+static int WT_CDECL
+__import_entry_cmp(const void *a, const void *b)
+{
+    WT_IMPORT_ENTRY *ae, *be;
+
+    ae = (WT_IMPORT_ENTRY *)a;
+    be = (WT_IMPORT_ENTRY *)b;
+
+    return strcmp(ae->name, be->name);
+}
+
+/*
+ * __schema_parse_wt_export --
+ *     Parse export metadata file and populate array of name/config entries. The array is sorted by
+ *     entry name. Caller is responsible to free any memory allocated for the import list.
+ */
+static int
+__schema_parse_wt_export(
+  WT_SESSION_IMPL *session, const char *export_file, WT_IMPORT_LIST *import_list)
+{
+    WT_DECL_ITEM(key);
+    WT_DECL_ITEM(value);
+    WT_DECL_RET;
+    WT_FSTREAM *fs;
+    char *metacfg;
+    const char *md_key, *md_value;
+    bool exist;
+
+    metacfg = NULL;
+
+    /* Look for a metadata export file: if we find it, load it. */
+    WT_RET(__wt_fs_exist(session, export_file, &exist));
+    if (!exist)
+        return (0);
+    WT_RET(__wt_fopen(session, export_file, 0, WT_STREAM_READ, &fs));
+
+    /* Read line pairs and add them to the import list. */
+    WT_ERR(__wt_scr_alloc(session, 1024, &key));
+    WT_ERR(__wt_scr_alloc(session, 1024, &value));
+    for (;;) {
+        WT_ERR(__wt_getline(session, fs, key));
+        if (key->size == 0)
+            break;
+        WT_ERR(__wt_getline(session, fs, value));
+        if (value->size == 0)
+            WT_ERR_PANIC(session, EINVAL, "%s: zero-length value", export_file);
+
+        /* Skip any WiredTiger tables and system entries. */
+        md_key = (const char *)key->data;
+        md_value = (const char *)value->data;
+        if (WT_SUFFIX_MATCH(md_key, WT_METAFILE) || WT_SUFFIX_MATCH(md_key, WT_HS_FILE) ||
+          WT_PREFIX_MATCH(md_key, WT_SYSTEM_PREFIX))
+            continue;
+
+        /*
+         * Check if the entry already exists in the metadata. Raise an error if we try to import an
+         * existing URI rather than just silently returning.
+         */
+        if ((ret = __wt_metadata_search(session, md_key, &metacfg)) != WT_NOTFOUND) {
+            WT_TRET(EEXIST);
+            goto err;
```

Comment:
Doesn't the create code itself handle existence checks? We should continue to let it do so.

Reply:
@sueloverso,  I actually wanted to ask you what to do if we find out that an entry already exists in the recipient's metadata. Should we fail or overwrite it? If we don't crash, at least we should let the caller know that we've overwritten the entry.

Reply:
The current create and/or import code should already handle that situation. It should be handled at another level. Check for a test for that to make sure. But even without `WiredTiger.export` someone today could try to import a URI that already exists in the database.

Reply:
OK, thank you!

Reply:
Done 960e78553238552b942e65ea99373da02a16e244 .